### DISCUSSION TITLE
Inconsistent tabular width between stacktrace and other outputs, e.g. the help.

### DISCUSSION THREAD

**Discussion by racinmat** [fastapi/typer#787] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

app = typer.Typer()


@app.command()
def create(username: str):
    print(f"Creating user: {username}")


@app.command()
def fail():
    print(f"Failing")
    42 / 0


if __name__ == "__main__":
    app()
```

### Description

The output of the command above is:
```
(wanna-ml-py310) PS C:\Projects\others\wanna-ml\my_tests> python .\testing_typer.py --help

 Usage: testing_typer.py [OPTIONS] COMMAND [ARGS]...

╭─ Options ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --install-completion          Install completion for the current shell.                                                                              │
│ --show-completion             Show completion for the current shell, to copy it or customize the installation.                                       │
│ --help                        Show this message and exit.                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Commands ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ create                                                                                                                                               │
│ fail                                                                                                                                                 │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

(wanna-ml-py310) PS C:\Projects\others\wanna-ml\my_tests> python .\testing_typer.py fail  
Failing
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Projects\others\wanna-ml\my_tests\testing_typer.py:14 in fail                                 │
│                                                                                                  │
│   11 @app.command()                                                                              │
│   12 def fail():                                                                                 │
│   13 │   print(f"Failing")                                                                       │
│ ❱ 14 │   42 / 0                                                                                  │
│   15                                                                                             │
│   16                                                                                             │
│   17 if __name__ == "__main__":                                                                  │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ZeroDivisionError: division by zero
```
note that the width of the table for help is much higher than the width of stacktrace. The max width for stack trace is only 100, which is very small, making it very hard to read for many our use-cases.
Specifically, the discrepancy is caused by this:
the help formatting is here https://github.com/tiangolo/typer/blob/0.12.0/typer/core.py#L674-L678 which calls https://github.com/tiangolo/typer/blob/0.12.0/typer/rich_utils.py#L126-L144 which sets the width to https://github.com/tiangolo/typer/blob/0.12.0/typer/rich_utils.py#L70 which is wither `TERMINAL_WIDTH` env var or None, leaving the `rich` package to figure it out.
While in the exception hook, the https://github.com/tiangolo/typer/blob/0.12.0/typer/main.py#L53-L102 is used which calls the `rich` from https://github.com/tiangolo/typer/blob/0.12.0/typer/main.py#L72-L79 and there it does not set the `width` so the default `100` is used, see https://github.com/Textualize/rich/blob/v13.7.1/rich/traceback.py#L245. And the console used by the the except hook is created in https://github.com/tiangolo/typer/blob/0.12.0/typer/main.py#L44 while it should instead obtain the rich console which contains the styling. This would make them more consistent.

This bug happens regardless of OS, although figuring out if the default showcase will be `80` or `100` is platform-dependent to some extent.

### Operating System

Windows

### Operating System Details

Windows 10

### Typer Version

0.12.0

### Python Version

Python 3.11.4

**Comment by tiangolo** [comment, marked as answer]
Thanks for the report and thanks for the fix in https://github.com/fastapi/typer/pull/788 :rocket: 

This will be available in Typer `0.12.5` released in the next few hours. :tada:
